install.sh

```diff
@@ -238,14 +238,24 @@ if [ $makerfaire2018 -eq 0 ]; then
   venv/bin/python -m snips_nlu generate-dataset fr */nlu/intent_fr.yaml > nabd/nlu/nlu_dataset_fr.json
 
   echo "Persisting Snips engines"
-  if [ -d nabd/nlu/engine_en ]; then
-    rm -rf nabd/nlu/engine_en
-  fi
-  venv/bin/snips-nlu train nabd/nlu/nlu_dataset_en.json nabd/nlu/engine_en
-  if [ -d nabd/nlu/engine_fr ]; then
-    rm -rf nabd/nlu/engine_fr
-  fi
-  venv/bin/snips-nlu train nabd/nlu/nlu_dataset_fr.json nabd/nlu/engine_fr
+  #
+  # Regroupping the logic for all the languages supported
+  for LANG in "en" "fr"; do
+    ENGINE="nabd/nlu/engine_${LANG}"
+    if [ -d "${ENGINE}" ]; then
+      #
+      file_age=$(stat -c %Y "${ENGINE}")
+      epoch_now=$(date +"%s")
+      #
+      #Trying to save some time & CPU in case the dataset was generated already today
+      if (( (epoch_now - file_age) > (60 * 60 * 24) )); then
+        rm -rf "${ENGINE}"
+        venv/bin/snips-nlu train nabd/nlu/nlu_dataset_${LANG}.json ${ENGINE}
+      fi
```

Comment:
This optimisation may not be desirable.
It may indeed seem overkill to re-train the NLU engine when the `install.sh` script is invoked repeatedly, for testing purposes or whatever.
But in the general use case (regular update) this optimisation will be a problem in the (unlikely but possible) case of 2 successive (less than 24h apart) updates  where the second update includes changes to NLU source data: this change will then be missed.

Reply:
The original scenario was to run the upgrade script after upgrading the OS and having it fail at different points (early adopter). You are right that 24h may be a bit generous. But the idea is to save CPU and time when running it again and again within a window of a few hours.
We could drop the threshold to 2h...

Reply:
> We could drop the threshold to 2h...

That would still leave a window for failure.

> But the idea is to save CPU and time when running it again and again within a window of a few hours.

Sure.
But if you want to have this "optimisation" in a public commit, you should at least condition it by the `install.sh` `test` parameter/flag.
The early adopter/tester would then set this flag when running `install.sh`.

Reply:
Fair enough. As implementing a new flag for the install.sh would be a decent PR on its own, I removed this change from the diff to keep only the efforts around the git commands.

I still have to apply the additional --no-rebase in the upgrade.sh but I keep it as a reason to push the CI to rebuild (as it is not very cooperative lately...).

Reply:
>  I removed this change from the diff to keep only the efforts around the git commands.

Fine. You can still keep the optimisation in a private commit, and just `git cherry-pick` it for your tests...

> .... to push the CI to rebuild (as it is not very cooperative lately...).

Well, for this there is not hope till issue #329 is fixed...